### PR TITLE
Remove the deprecated load_usgs_quakes function (deprecated since v0.6.0)

### PR DESCRIPTION
**Pull request by willschlitzer**
This replaces `load_usgs_quakes` with the internal function `_load_usgs_quakes`. It also adds additional checks to `test_usgs_quakes`.

Addresses: #2302

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -215,36 +215,18 @@ def load_sample_bathymetry(**kwargs):
...
     """
-    (Deprecated) Load a table of global earthquakes from the USGS as a
-    pandas.DataFrame.
-
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="usgs_quakes")`` and will be removed in
-       v0.9.0.
+    Load a table of global earthquakes from the USGS as a pandas.DataFrame.
 
     This is the ``@usgs_quakes_22.txt`` dataset used in the GMT tutorials.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:222]: Please see the comments at https://github.com/GenericMappingTools/pygmt/pull/2301#discussion_r1062048779.
```suggestion
```

File: pygmt/tests/test_datasets_samples.py [hunk 2]
```diff
@@ -72,10 +71,56 @@ def test_usgs_quakes():
     """
     Check that the dataset loads without errors.
```
Comment (yvonnefroehlich) [inline pygmt/tests/test_datasets_samples.py:72]:
```suggestion
    Check that the @usgs_quakes_22.txt dataset loads without errors.
```

File: pygmt/datasets/samples.py [hunk 3]
```diff
@@ -198,36 +198,16 @@ def load_sample_bathymetry(**kwargs):
...
-        warnings.warn(
-            "This function has been deprecated since v0.6.0 and will be "
-            "removed in v0.9.0. Please use "
-            "load_sample_data(name='usgs_quakes') instead.",
-            category=FutureWarning,
-            stacklevel=2,
-        )
     fname = which("@usgs_quakes_22.txt", download="c")
     data = pd.read_csv(fname)
     return data
```
Comment (seisman) [inline pygmt/datasets/samples.py:213]:
```suggestion
    return pd.read_csv(fname)
```

**Comment by seisman** [comment]
I just realized that this sample dataset is not used in any examples. Should we remove it?

**Comment by yvonnefroehlich** [comment]
> I just realized that this sample dataset is not used in any examples. Should we remove it?

Hm. Not sure, as there seem to be other sample datasets which are currently also not used in any example, e.g., `maunaloa_co2`, `hotspots`, or `mars_shape`.

**Comment by seisman** [comment]
> > I just realized that this sample dataset is not used in any examples. Should we remove it?
> 
> Hm. Not sure, as there seem to be other sample datasets which are currently also not used in any example, e.g., `maunaloa_co2`, `hotspots`, or `mars_shape`.

- `mars_shape` is used in `pygmt/src/sphinterpolate.py`, 
- `maunaloa_co2` is used in `pygmt/tests/test_filter1d.py`
- `hotspots` was added in https://github.com/GenericMappingTools/pygmt/pull/1386, and it seems there was a plan to use it in a sphdistance example.

**Comment by willschlitzer** [comment]
I'm fine with removing `load_usgs_quakes`. I don't remember why (or when) I added it.

**Comment by weiji14** [comment]
A quick search on GitHub shows it being used before at https://github.com/GenericMappingTools/try-gmt-python/blob/master/try-gmt-python.ipynb. But it's 4 years old now.

How much of a burden is it to keep the function maintained? Maybe we could just keep it in case?

**Comment by willschlitzer** [comment]
It's not too much of an issue to keep it maintained. I just assumed that @seisman suggested removing it to avoid the precedent that all GMT remote files have an associated PyGMT function to load them.

**Comment by seisman** [comment]
> It's not too much of an issue to keep it maintained. I just assumed that @seisman suggested removing it to avoid the precedent that all GMT remote files have an associated PyGMT function to load them.

I'm OK with maintaining this function, and I believe this dataset for global earthquakes can be used in examples in the future.

My major concern is that the returned data table contains too many columns and most of them are  text strings and are unlikely used in any PyGMT examples. I suggest only keeping the "time", "latitude", "longitude", "depth" and "mag", similar to the data table returned by `_load_japan_quakes`.

**Comment by willschlitzer** [comment]
> > It's not too much of an issue to keep it maintained. I just assumed that @seisman suggested removing it to avoid the precedent that all GMT remote files have an associated PyGMT function to load them.
> 
> I'm OK with maintaining this function, and I believe this dataset for global earthquakes can be used in examples in the future.
> 
> My major concern is that the returned data table contains too many columns and most of them are text strings and are unlikely used in any PyGMT examples. I suggest only keeping the "time", "latitude", "longitude", "depth" and "mag", similar to the data table returned by `_load_japan_quakes`.

I've never studied seismology, so I'm not sure how important information that pertains to the errors is to someone hoping to use this dataset. But I don't see the need to load this dataset with a separate function if we're going to remove quite a few of the columns. If it's the same columns as `_load_japan_quakes`, we might as well just use the Japan quakes dataset for any example/test functions. 

Looking at the [USGS Earthquake API](https://earthquake.usgs.gov/fdsnws/event/1/), we could probably create an example that incorporates getting a JSON file from the API and using it to create a plot/perform operations. I think that would be a better example of a real application of earthquake data instead of  `usgs_quakes_22.txt`.

File: pygmt/datasets/samples.py [hunk 4]
```diff
@@ -73,6 +73,7 @@ def load_sample_data(name):
         "bathymetry": load_sample_bathymetry,
         "hotspots": load_hotspots,
         "mars_shape": load_mars_shape,
+        "ocean_ridge_points": load_ocean_ridge_points,
```
Comment (seisman) [inline pygmt/datasets/samples.py:76]:
```suggestion
```

**Comment by seisman** [comment]
> > > It's not too much of an issue to keep it maintained. I just assumed that @seisman suggested removing it to avoid the precedent that all GMT remote files have an associated PyGMT function to load them.
> > 
> > 
> > I'm OK with maintaining this function, and I believe this dataset for global earthquakes can be used in examples in the future.
> > My major concern is that the returned data table contains too many columns and most of them are text strings and are unlikely used in any PyGMT examples. I suggest only keeping the "time", "latitude", "longitude", "depth" and "mag", similar to the data table returned by `_load_japan_quakes`.
> 
> I've never studied seismology, so I'm not sure how important information that pertains to the errors is to someone hoping to use this dataset. But I don't see the need to load this dataset with a separate function if we're going to remove quite a few of the columns. If it's the same columns as `_load_japan_quakes`, we might as well just use the Japan quakes dataset for any example/test functions. I don't feel strongly about removing this function, as someone may already be using it in their examples, but I don't wish to pare down the information in the returned dataset.

OK, I think it's fine to keep this function unchanged.

> Looking at the [USGS Earthquake API](https://earthquake.usgs.gov/fdsnws/event/1/), we could probably create an example that incorporates getting a JSON file from the API and using it to create a plot/perform operations. I think that would be a better example of a real application of earthquake data instead of `usgs_quakes_22.txt`.

Seismologists usually use [ObsPy's get_events() function](https://docs.obspy.org/packages/autogen/obspy.clients.fdsn.client.Client.get_events.html#obspy.clients.fdsn.client.Client.get_events) to get earthquakes from USGS or other seismic data centers. The function returns a [`Catalog` object](https://docs.obspy.org/packages/autogen/obspy.core.event.Catalog.html) which contains event information. This is related to the ObsPy integration in https://github.com/GenericMappingTools/pygmt/issues/967.

File: pygmt/datasets/samples.py [hunk 5]
```diff
@@ -85,6 +85,7 @@ def load_sample_data(name):
         "notre_dame_topography": _load_notre_dame_topography,
         "ocean_ridge_points": _load_ocean_ridge_points,
         "rock_compositions": _load_rock_sample_compositions,
+        "usgs_quakes": _load_usgs_quakes,
```
Comment (seisman) [inline pygmt/datasets/samples.py:88]: Need to remove `"usgs_quakes": load_usgs_quakes` at line 75.

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**